examples/pipeline_tacotron2/inference.py

```diff
@@ -0,0 +1,320 @@
+    r"""torch.distributed.DistributedDataParallel wraps the model with an additional "module.".
+    This function unwraps this layer so that the weights can be loaded on models with a single GPU.
+
+    Args:
+        state_dict: Original state_dict.
+
+    Return:
+        unwrapped_state_dict: Unwrapped state_dict.
+    """
+
+    return {k.replace('module.', ''): v for k, v in state_dict.items()}
+
+
+def main(args):
+    torch.manual_seed(0)
+    random.seed(0)
+    np.random.seed(0)
+
+    device = "cuda" if torch.cuda.is_available() else "cpu"
+
+    if args.checkpoint_path is None and args.checkpoint_name is None:
+        raise ValueError("Either --checkpoint-path or --checkpoint-name must be specified.")
+    elif args.checkpoint_path is not None and args.checkpoint_name is not None:
+        raise ValueError("Both --checkpoint-path and --checkpoint-name are specified, "
+                         "can only specify one.")
+
+    n_symbols = len(get_symbol_list(args.text_preprocessor))
+    text_preprocessor = partial(
+        text_to_sequence,
+        symbol_list=args.text_preprocessor,
+        phonemizer=args.phonemizer,
+        checkpoint=args.phonemizer_checkpoint,
+        cmudict_root=args.cmudict_root,
+    )
+
+    if args.checkpoint_path is not None:
+        tacotron2 = Tacotron2(n_symbol=n_symbols)
+        tacotron2.load_state_dict(
+            unwrap_distributed(torch.load(args.checkpoint_path, map_location=device)['state_dict']))
+        tacotron2 = tacotron2.to(device).eval()
+    elif args.checkpoint_name is not None:
+        tacotron2 = pretrained_tacotron2(args.checkpoint_name).to(device).eval()
+
+        if n_symbols != tacotron2.n_symbols:
+            raise ValueError("the number of symbols for text_preprocessor ({n_symbols}) "
+                             "should match the number of symbols for the"
+                             "pretrained tacotron2 ({tacotron2.n_symbols}).")
+
+    if args.jit:
+        tacotron2 = torch.jit.script(tacotron2)
+
+    sequences, lengths = prepare_input_sequence([args.input_text],
+                                                text_processor=text_preprocessor)
+    sequences, lengths = sequences.long().to(device), lengths.long().to(device)
+    with torch.no_grad():
+        mel_specgram, _, _ = tacotron2.infer(sequences, lengths)
+
+    if args.vocoder == "nvidia_waveglow":
+        waveglow = torch.hub.load('NVIDIA/DeepLearningExamples:torchhub', 'nvidia_waveglow', model_math='fp16')
+        waveglow = waveglow.remove_weightnorm(waveglow)
+        waveglow = waveglow.to(device)
+        waveglow.eval()
+
+        if args.jit:
+            raise ValueError("Vocoder option `nvidia_waveglow is not jittable.")
+
+        with torch.no_grad():
+            waveform = waveglow.infer(mel_specgram).cpu()
+
+    elif args.vocoder == "wavernn":
+        from torchaudio.models import wavernn
+        sys.path.append(os.path.join(os.path.dirname(__file__), "../pipeline_wavernn"))
```

Comment:
Not in the scope of this PR, but I think putting components for wavernn and tacotoron2 under the same directory, say `examples/tts` would allow to organize the code better.

Reply:
I've created [an issue](https://github.com/pytorch/audio/issues/1712) to keep track of this.